### PR TITLE
chore(menu): refactor (VIV-1204)

### PR DESCRIPTION
**Pull request by YonatanKra**
Follow the commits

### CODE REVIEW

File: libs/components/src/lib/menu/menu.spec.ts [hunk 1]
```diff
@@ -109,35 +85,31 @@ describe('vwc-menu', () => {
...
+	describe('focus', () => {
+		function createMenuItem(type = 'menuitem') {
 			const div = document.createElement('div');
-			div.setAttribute('role', 'menuitem');
+			div.setAttribute('role', type);
 			element.appendChild(div);
+			return div;
+		}
+		it('should focus the first menuitem in the menu', async () => {
+			const div = createMenuItem();
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.spec.ts:112]: the fact that this is in the HTML does not test that it is focused. Checking the menu manually discovers that there's no way to get to the menu items if you don't have hands and eyes :) :(

File: libs/components/src/lib/menu/menu.template.ts [hunk 2]
```diff
@@ -24,16 +24,17 @@ export const MenuTemplate: (
 ) => ElementViewTemplate = (context: ElementDefinitionContext) => {
 	const popupTag = context.tagFor(Popup);
 
+	function handlePopupEvents(x: Menu, state: boolean) {
+		x.open = state;
+	}
 	return html<Menu>`
-		<template
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.template.ts:28]: 👍

**Review by rachelbt — COMMENTED** [review]
nice idea of commits as notes (though I like also simple bullets in the PR comments :) )

File: libs/components/src/lib/menu/menu.spec.ts (hunk 1 above)
Comment (YonatanKra) [reply libs/components/src/lib/menu/menu.spec.ts:112]: It doesn't check it is in the HTML. It checks that this is the focused (`document.activeElement`).

**Review by rinaok — APPROVED** [review]

**Review by rachelbt — APPROVED** [review]
LGTM

**Pull request merged**